src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -49,54 +50,174 @@ nodeExtractText = mconcat . map extractText . nodeFlatten
         CODE t -> t
         _ -> ""
 
-nodeExtractInfo :: Node -> ExceptT Text Identity FileInfo
-nodeExtractInfo docNode = fmap finaliseFileInfo $ execStateT (loop docNode) def
-  where
-    loop node@(Node pos ty subs) = case ty of
-        DOCUMENT ->
-            mapM_ loop subs
-        PARAGRAPH ->
-            mapM_ loop subs
-        HEADING lvl ->
-            let text = nodeExtractText node
-                aType = HeaderAnchor lvl
-                aName = headerToAnchor text
-                aPos = toPosition pos
-            in fiAnchors %= (Anchor{..} :)
-        LIST _ ->
-            mapM_ loop subs
-        ITEM ->
-            mapM_ loop subs
-        HTML_INLINE htmlText -> do
-            let mName = T.stripSuffix "\">" =<< T.stripPrefix "<a name=\"" htmlText
-            whenJust mName $ \aName -> do
-                let aType = HandAnchor
-                    aPos = toPosition pos
-                fiAnchors %= (Anchor{..} :)
-        LINK url _ -> do
-            let rName = nodeExtractText node
-                rPos = toPosition pos
-                link = if null url then rName else url
-            let (rLink, rAnchor) = case T.splitOn "#" link of
-                    [t]    -> (t, Nothing)
-                    t : ts -> (t, Just $ T.intercalate "#" ts)
-                    []     -> error "impossible"
-            fiReferences %= (Reference{..} :)
-        _ -> pass
-
-parseFileInfo :: FilePath -> LT.Text -> FileInfo
-parseFileInfo path input =
-    let outcome = runIdentity . runExceptT $
-                  nodeExtractInfo $ commonmarkToNode [] [] $ toStrict input
-    in case outcome of
-        Left err  -> error $ "Failed to parse file " <> show path <>
-                             ": " <> show err
-        Right res -> res
+data IgnoreMode
+    = Link
+    | Paragraph
+    | File
+    deriving Eq
+
+nodeExtractInfo :: Node -> Except Text FileInfo
+nodeExtractInfo (Node _ _ docNodes) =
+    if checkIgnoreFile docNodes
+    then return def
+    else finaliseFileInfo <$> extractionResult
+    where
+        extractionResult :: Except Text FileInfo
+        extractionResult =
+            let exceptRes = runExceptT $ loop docNodes Nothing
+                (eitherError, fileInfo) = runState exceptRes def
+            in except $ either Left (const $ Right fileInfo) eitherError
```

Comment:
> > If you don't want any link in your repository to be ignored, leave `ignoreRefs` empty:
> 
> Can `ignoreRefs` be omitted?

[Now](https://github.com/serokell/xrefcheck/pull/53/commits/d40f57a78ef63cf5bf65d552461c8a5b3eddb499) it can be